### PR TITLE
Buyer-Apartment-Locality-Model-Dao-Service-Test-Added

### PR DESCRIPTION
**Pull request by SainathB**
Buyer-Apartment-Locality-Model-Dao-Service-Test-Added

### CODE REVIEW

File: src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+		catch (Exception exception) {
+			try {
+				if (transaction != null)
+					transaction.rollback();
+				throw TransactionFailureException;
+			}
+
+			catch (RuntimeException runtimeException) {
+				throw RollbackFailureException;
```
Comment (sairamch04) [inline src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java:164]: Here the  Transaction rollback failed exception is thrown even if The transaction rollback is successfull

File: src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+			apartmentDAO.insert(apartment);
+			transaction.commit();
+			return apartment;
+		}
+
+		catch (Exception exception) {
+			try {
+				if (transaction != null)
+					transaction.rollback();
+				throw TransactionFailureException;
```
Comment (sairamch04) [inline src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java:160]: Same type of exception is thrown even if rollback is successfull or not which should not be the case

File: src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+			ApartmentDAO apartmentDao = new ApartmentDAOImpl(session);
+			apartmentDao.delete(apartment);
+			transaction.commit();
+		}
+
+		catch (Exception exception) {
+			try {
+				if (transaction != null)
+					transaction.rollback();
```
Comment (sairamch04) [inline src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java:274]: Same here and in all functions.

File: src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java (hunk 1 above)
Comment (SainathB) [inline src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java:164]: If transaction is a failure and rollback is successful then TransactionFailureException is thrown.
If transaction is a failure and rollback is a failure then RollbackFailureException is thrown.
This case was handled in the same way in the snippet sent by Deepak.

Comment (sairamch04) [inline src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java:164]: No. See that snippet again. IF ELSE  was not written . Also look what happens when exception is thrown

File: src/main/java/com/foodybuddy/dao/impl/LocalityDAOImpl.java [hunk 4]
```diff
@@ -1,46 +1,87 @@
...
 
+	/* (non-Javadoc)
+	 * @see com.foodybuddy.dao.LocalityDAO#getAll()
+	 */
 	@SuppressWarnings("unchecked")
 	public List<Locality> getAll() {
 		String query = "FROM Locality";
-		List<Locality> localitiesList = session.createQuery(query).list();
-		return localitiesList;
+		List<Locality> citiesList = this.session.createQuery(query).list();
```
Comment (kanmaytacker) [inline src/main/java/com/foodybuddy/dao/impl/LocalityDAOImpl.java:61]: Ambiguous variable name citiesList.

**Pull request merged**